SRR2016686 | predicted serotype:	bispebjergnote:	 | abortusequi
SRR2016688 | predicted serotype:	bispebjergnote:	 | abortusequi
ERR230405 | predicted serotype:	bispebjergnote:	 | abortusequi
SRR11101179 | predicted serotype:	i -:f,g:-note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | adelaide
SRR1643064 | predicted serotype:	agonote:	 | aderike
SRR9994447 | predicted serotype:	typhimuriumnote:	 | albany
SRR5215710 | predicted serotype:	i 35:g,s,t:1,5note:	this predicted serotype is not in the kauffman-white scheme.  | anecho
SRR1917438 | predicted serotype:	iv -:z4,z23:-note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | baguida
SRR1805579 | predicted serotype:	iv -:z4,z23:-note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | baguida
SRR1264989 | predicted serotype:	i 21:z4,z23:e,n,z15note:	this predicted serotype is not in the kauffman-white scheme.  | baguida
SRR498275 | predicted serotype:	i -:y:1,5note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | bareilly
SRR2015814 | predicted serotype:	orionnote:	 | binza
SRR2131384 | predicted serotype:	orionnote:	 | binza
SRR1248802 | predicted serotype:	orionnote:	 | binza
SRR1118697 | predicted serotype:	tshiongwenote:	 | braenderup
SRR6742490 | predicted serotype:	agonanote:	 | caracas
SRR6325064 | predicted serotype:	heidelbergnote:	 | chester
SRR8550552 | predicted serotype:	cubananote:	 | coeln
SRR5047529 | predicted serotype:	i 7:l,v:l,wnote:	this predicted serotype is not in the kauffman-white scheme.  | coleypark
SRR2533427 | predicted serotype:	i 7:z4,z24:l,wnote:	this predicted serotype is not in the kauffman-white scheme.  | coleypark
SRR8266798 | predicted serotype:	agamanote:	 | colindale
SRR8267457 | predicted serotype:	virchownote:	 | colindale
SRR9994446 | predicted serotype:	typhimuriumnote:	 | colindale
SRR8159839 | predicted serotype:	i -:l,v:1,2note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | concord
SRR8159842 | predicted serotype:	i -:l,v:1,2note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | concord
SRR8468685 | predicted serotype:	i -:f,g:-note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | derby
SRR3384027 | predicted serotype:	kottbusnote:	 | eastbourne
SRR5105538 | predicted serotype:	brazzavillenote:	 | edinburg
SRR5105539 | predicted serotype:	brazzavillenote:	 | edinburg
SRR7516573 | predicted serotype:	livingstonenote:	 | eimsbuettel
SRR7501171 | predicted serotype:	livingstonenote:	 | eimsbuettel
SRR7480352 | predicted serotype:	livingstonenote:	 | eimsbuettel
SRR6916429 | predicted serotype:	i -:k:1,2note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | galiema
SRR1122517 | predicted serotype:	braenderupnote:	 | gallinarum
SRR5975634 | predicted serotype:	i -:d:1,7note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | gaminara
SRR5216353 | predicted serotype:	claiborneinote:	 | goverdhan
SRR7187112 | predicted serotype:	claiborneinote:	 | goverdhan
SRR8554048 | predicted serotype:	claiborneinote:	 | goverdhan
SRR1604876 | predicted serotype:	typhimuriumnote:	 | indiana
SRR8329372 | predicted serotype:	paratyphi bnote:	  | java
SRR1604873 | predicted serotype:	paratyphi bnote:	  | java
SRR1106380 | predicted serotype:	paratyphi bnote:	  | java
SRR6950286 | predicted serotype:	kingstonnote:	 | kintambo
SRR10843966 | predicted serotype:	coelnnote:	 | kisangani
SRR2124442 | predicted serotype:	i 47:z4,z23:-note:	this predicted serotype is not in the kauffman-white scheme. this antigenic profile has been associated with serotype 'iiia 47:z4,z23:-' in the kauffman-white scheme. the existence of the same antigenic formula in multiple species or subspecies is well documented in the kauffman-white scheme.  | kua
SRR2085740 | predicted serotype:	i 47:z4,z23:-note:	this predicted serotype is not in the kauffman-white scheme. this antigenic profile has been associated with serotype 'iiia 47:z4,z23:-' in the kauffman-white scheme. the existence of the same antigenic formula in multiple species or subspecies is well documented in the kauffman-white scheme.  | kua
SRR1481653 | predicted serotype:	kokomlemlenote:	 | kua
SRR1967309 | predicted serotype:	typhimuriumnote:	 | lagos
SRR6207640 | predicted serotype:	mbandakanote:	 | lille
SRR7028169 | predicted serotype:	i -:k:e,n,xnote:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | matadi
SRR1122614 | predicted serotype:	hadarnote:	 | mbandaka
SRR7130344 | predicted serotype:	schwarzengrundnote:	 | minnesota
SRR1106338 | predicted serotype:	manhattannote:	 | muenchen
SRR1106350 | predicted serotype:	muenchennote:	 | muenster
SRR1106344 | predicted serotype:	i 7:g,m,s:1,5note:	this predicted serotype is not in the kauffman-white scheme.  | montevideo
SRR1171953 | predicted serotype:	i 8:g,m:1,2note:	this predicted serotype is not in the kauffman-white scheme.  | newport
SRR1106441 | predicted serotype:	paratyphi bnote:	  | oranienburg
SRR3229343 | predicted serotype:	poonanote:	 | oakland
SRR3057221 | predicted serotype:	anatumnote:	 | norwich
SRR4896325 | predicted serotype:	saphranote:	 | nima
ERR2115652 | predicted serotype:	i 2:l,z13:e,n,xnote:	this predicted serotype is not in the kauffman-white scheme.  | napoli
SRR8924894 | predicted serotype:	i 2:l,z13:e,n,xnote:	this predicted serotype is not in the kauffman-white scheme.  | napoli
SRR7646078 | predicted serotype:	i 2:l,z13:e,n,xnote:	this predicted serotype is not in the kauffman-white scheme.  | napoli
SRR1449925 | predicted serotype:	enteritidisnote:	detected sdf i that is characteristic of commonly circulating strains of serotype enteritidis.  | ramatgan
SRR12664401 | predicted serotype:	gallinarum or enteritidisnote:	sdf i that is characteristic of commonly circulating strains of serotype enteritidis was not detected. the predicted serotypes share the same general formula: 9:g,m:- and can be differentiated by additional analysis.  | pullorum
SRR8655899 | predicted serotype:	enteritidisnote:	detected sdf i that is characteristic of commonly circulating strains of serotype enteritidis.  | pullorum
SRR2016689 | predicted serotype:	gallinarum or enteritidisnote:	sdf i that is characteristic of commonly circulating strains of serotype enteritidis was not detected. the predicted serotypes share the same general formula: 9:g,m:- and can be differentiated by additional analysis.  | pullorum
SRR1060638 | predicted serotype:	limetenote:	 | paratyphi c
SRR6208799 | predicted serotype:	i -:k:e,n,z15note:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | orientalis
SRR1106438 | predicted serotype:	typhimuriumnote:	 | panama
SRR5819981 | predicted serotype:	- -:-:-note:	the input genome cannot be identified as salmonella. check the input for taxonomic id, contamination, or sequencing quality.  | rissen
SRR3038247 | predicted serotype:	bareillynote:	 | richmond
SRR5040796 | predicted serotype:	anatumnote:	 | richmond
SRR12842636 | predicted serotype:	kentuckynote:	 | reading
SRR955194 | predicted serotype:	gallinarum or enteritidisnote:	sdf i that is characteristic of commonly circulating strains of serotype enteritidis was not detected. the predicted serotypes share the same general formula: 9:g,m:- and can be differentiated by additional analysis.  | stanleyville
SRR1106426 | predicted serotype:	ebokonote:	 | sandiego
SRR8896362 | predicted serotype:	chesternote:	 | sandiego
SRR1182727 | predicted serotype:	ii 42:b:e,n,x,z15note:	 | tomegbe
SRR3721577 | predicted serotype:	ii 42:b:e,n,x,z15note:	 | tomegbe
SRR2086557 | predicted serotype:	ii 42:b:e,n,x,z15note:	 | tomegbe
SRR1772987 | predicted serotype:	milwaukeenote:	 | treforest
SRR1060703 | predicted serotype:	i -:b:e,n,xnote:	o antigen was not detected. this result may be due to a rough strain that has deleted the rfb region. for raw reads input, the k-mer workflow is sometimes more sensitive than the microassembly workflow in detecting o antigen. caution should be used with this approach because the k-mer result may be due to low levels of contamination.  | urbana
SRR8526059 | predicted serotype:	cochinnote:	 | zanzibar
SRR6951171 | predicted serotype:	senftenbergnote:	 | westhampton
SRR2124283 | predicted serotype:	senftenbergnote:	 | westhampton
414 499 0.8296593186372746
